### PR TITLE
Updated user manual

### PR DESCRIPTION
**Pull request by AmalieDue**
### Description
The Encryptonize Core user manual has been updated.

Closes DEV-135

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [x] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [ ] I have run `make lint`.
- [ ] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [x] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [ ] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: documentation/user_manual.md [hunk 1]
```diff
@@ -132,75 +118,163 @@ E.g., to create a user with `READ` and `CREATE` scopes, call `./encryption-servi
...
 request should contain an attribute named `scopes` which enumerates all the scopes the user's
 initial group should have. Users can only be created by a user with the `USERMANAGEMENT` scope.
 
 Once a user has been created, a new `user_id` and `password` will be returned from the call. Note
 that a group with the requested scopes and an ID equal to the `user_id` is automatically created.
 
+#### Example with the CreateUser endpoint
+Input:
+```bash
+grpcurl -plaintext -H "authorization: bearer <access token>"
```
Comment (philip-cc) [inline documentation/user_manual.md:145]: We should at a minimum have a link that grpcurl for people who don't know what that is.

File: documentation/user_manual.md [hunk 2]
```diff
@@ -132,75 +118,163 @@ E.g., to create a user with `READ` and `CREATE` scopes, call `./encryption-servi
...
+In order to modify the members of a group, you need to call the `encryptonize.Authn.AddUserToGroup`
+and `encryptonize.Authn.RemoveUserFromGroup` endpoints. In both cases the request should contain the
 `group_id` of the group in question and the `user_id` of the user to be added/removed.
 
 ## Storage-less encryption
 
-The Encryptonize API allows to bypass the storage and instead return the encrypted packages back to
-the user. This might be useful if you wish to manage the encrypted data yourself.
+The Encryptonize API does not provide storage, so the encrypted packages are sent back to
+the user.
```
Comment (philip-cc) [inline documentation/user_manual.md:224]: Mentioning storage here doesn't really make much sense as there is nothing to contrast it to anymore.

File: documentation/user_manual.md [hunk 3]
```diff
@@ -132,75 +118,163 @@ E.g., to create a user with `READ` and `CREATE` scopes, call `./encryption-servi
...
 
 ### Encryption
 
-You can encrypt an object and get the encrypted package back using the `enc.Encryptonize.Encrypt
-endpoint`. The caller needs the `CREATE` scope in order to use this endpoint. Similar to the `Store`
-endpoint, you need to provide the `plaintext` and the `associated_data`. The response of this call
+You can encrypt an object and get the encrypted package back using the `encryptonize.Core.Encrypt`
+endpoint. The caller needs the `CREATE` scope in order to use this endpoint. You need to provide the `plaintext` and the `associated_data`. The response of this call
 will contain the `ciphertext`, the `associated_data` and the `object_id`. Note that the
 `associated_data` is not encrypted.
```
Comment (philip-cc) [inline documentation/user_manual.md:231]: We should probably mention that the AD *is* authenticated though.

File: documentation/user_manual.md [hunk 4]
```diff
@@ -132,75 +118,163 @@ E.g., to create a user with `READ` and `CREATE` scopes, call `./encryption-servi
...
 
 ### Decryption
 
-To decrypt an object, you need to call the `enc.Encryptonize.Decrypt` endpoint and provide the
+To decrypt an object, you need to call the `encryptonize.Core.Decrypt` endpoint and provide the
 `ciphertext`, `associated_data` and the `object_id` in the request. This endpoint requires the
 `READ` scope. If you are authenticated towards the API and authorized to read the object, the
 response will contain the `plaintext` and the `associated_data`.
 
+#### Example with the Encrypt and Decrypt endpoints
```
Comment (philip-cc) [inline documentation/user_manual.md:240]: This should probably be split into two parts and nested under their respective sections.

File: documentation/user_manual.md [hunk 5]
```diff
@@ -1,18 +1,14 @@
...
 
-This user manual is updated to correspond to version 3.2.0 of the API.
+This user manual is updated to correspond to version 0.1.29 of the API.
 
-This document will provide a quick introduction on how to use the Encryptonize service and introduce
-essential concepts and terminology. For a detailed description of the gRPC API, see [the
-specification](../api/api-v3.2.0.md).
+This document will provide a quick introduction on how to use the Encryptonize service and introduce essential concepts and terminology. For a detailed description of the gRPC API, see [the specification](api.md).
 
 - [Overview](#overview)
```
[the remark this answers is not in the document]
Comment (AmalieDue) [reply documentation/user_manual.md:7]: FIxed

File: documentation/user_manual.md [hunk 6]
```diff
@@ -47,11 +43,11 @@ behaviour can be modified by setting the environment variable `ECTNZ_CONFIGFILE`
 formats are TOML, YAML, and JSON.
```
[the remark this answers is not in the document]
Comment (AmalieDue) [reply documentation/user_manual.md:29]: Fixed

File: documentation/user_manual.md [hunk 7]
```diff
@@ -91,7 +77,7 @@ Note that the access token is short lived (1 hour).
 
 ## Users and Groups
 
-A **user** is an authentication entity in the encryption server, and is only represented by a user
+A **user** is an authentication entity in Encryptonize, and is only represented by a user
```
[the remark this answers is not in the document]
Comment (AmalieDue) [reply documentation/user_manual.md]: Fixed

File: documentation/user_manual.md (hunk 1 above)
Comment (AmalieDue) [reply documentation/user_manual.md:123]: Fixed

File: documentation/user_manual.md (hunk 2 above)
Comment (AmalieDue) [reply documentation/user_manual.md]: Fixed

File: documentation/user_manual.md (hunk 3 above)
Comment (AmalieDue) [reply documentation/user_manual.md]: Fixed

File: documentation/user_manual.md (hunk 4 above)
Comment (AmalieDue) [reply documentation/user_manual.md]: Fixed

File: documentation/user_manual.md [hunk 8]
```diff
@@ -1,101 +1,64 @@
...
-This user manual is updated to correspond to version 3.2.0 of the API.
+This user manual is updated to correspond to version 0.1.29 of the API.
 
-This document will provide a quick introduction on how to use the Encryptonize service and introduce
-essential concepts and terminology. For a detailed description of the gRPC API, see [the
-specification](../api/api-v3.2.0.md).
+This document will provide a quick introduction on how to use Encryptonize Core and introduce essential concepts and terminology. For a detailed description of the gRPC API, see [the specification](api.md).
 
 - [Overview](#overview)
   - [Terminology](#terminology)
```
[the remark this answers is not in the document]
Comment (AmalieDue) [reply documentation/user_manual.md]: Fixed

**Review by philip-cc — APPROVED** [review]
Awesome!

**Pull request merged**